core/tests/puppeteer-acceptance-tests/utilities/user/logged-out-user.ts

```diff
@@ -1449,6 +1423,112 @@ export class LoggedOutUser extends BaseUser {
+   */
+  async clickPartnerWithUsButtonInAboutPage(): Promise<void> {
+    const allowedUrls = [
+      partnershipsFormShortUrl,
+      partnershipsFormUrl,
+      `${partnershipsFormUrl}?usp=send_form`,
+    ];
+
+    const partnerTab = this.isViewportAtMobileWidth()
+      ? partnerMobileTabInAboutPage
+      : partnerDesktopTabInAboutPage;
+    await this.clickOn(partnerTab);
+
+    const partnerWithUsButtonInAboutPage = this.isViewportAtMobileWidth()
+      ? partnerWithUsMobileButtonInAboutPage
+      : partnerWithUsDesktopButtonInAboutPage;
+
+    // The Google Form URL changes from the 1st to the 2nd and from 2nd to the
+    // 3rd in a short span of 500-1000 ms for it's own reasons which we can't
+    // control.So we need to check for all the 3 URLs as all of them are valid.
+    await this.clickLinkButtonToNewTabAndVerifyAllowedUrls(
+      partnerWithUsButtonInAboutPage,
+      'Partner With Us button at the bottom of the Partnerships page',
+      allowedUrls,
+      'Partnerships Google Form'
+    );
+  }
+
+  /**
+   * Function to click the Partner With Us button in the Partnerships page
+   * and check if it opens the Partnerships Google form in Portuguese.
+   */
+  async clickPartnerWithUsButtonInAboutPageInGivenLanguage(
+    langCode: string
+  ): Promise<void> {
+    await this.changeSiteLanguage(langCode);
+    // Here we need to reload the page again to confirm the language change.
+    await this.page.reload();
+
+    const partnerTab = this.isViewportAtMobileWidth()
+      ? partnerMobileTabInAboutPage
+      : partnerDesktopTabInAboutPage;
+    await this.clickOn(partnerTab);
+
+    const partnerWithUsButtonInAboutPage = this.isViewportAtMobileWidth()
+      ? partnerWithUsMobileButtonInAboutPage
+      : partnerWithUsDesktopButtonInAboutPage;
+
+    // Here we are not verifying the 3 URLs as we did in the English version
+    // because we have put the direct translated Google Form URL in the page itself.
+    // Refer core/templates/pages/partnerships-page/partnerships-page.component.ts to see how it's done.
+    await this.clickLinkButtonToNewTab(
+      partnerWithUsButtonInAboutPage,
+      'Partner With Us button at the bottom of the Partnerships page',
+      partnershipsFormInPortugueseUrl,
+      'Partnerships Google Form'
+    );
+  }
+
+  /**
+   * Function to check if the donor box is visible by clicking on the "Donate" button
+   * on the about page. Here we don't test the functionality of the donor box, just
+   * its visibility, because the donor box is an iframe and a third-party service.
+   */
+  async clickDonateButtonInAboutPage(): Promise<void> {
+    const donorTab = this.isViewportAtMobileWidth()
+      ? donorMobileTabInAboutPage
+      : donorDesktopTabInAboutPage;
+    await this.clickOn(donorTab);
+
+    const donateButtonInAboutPage = this.isViewportAtMobileWidth()
+      ? donateMobileButtonInAboutPage
+      : donateDesktopButtonInAboutPage;
+    await this.clickOn(donateButtonInAboutPage);
```

Comment:
Lets keep constant donarTab and donateButtonInAboutPage together just below function declaration and then start clicking and doing other operations for clearer code.